core/runner.py

```diff
@@ -221,3 +242,99 @@ def _initialize_model(self, job):
                     error_message="",
                 )
         return results
+
+
+    def _transform_and_split_data(self, job):
+        """
+        Takes in a job, which should have the raw filepath assigned.
+
+        1. In each folder, aggregate all data.
+        2. Create session folder in transformed folder, along with data folders
+           in session folder.
+        3. Transform data in each raw data folder.
+        4. Shuffle each transformed data and perform train-test split on each. 
+        5. Put each training and test set in corresponding data folders in 
+           session folder.
+        6. Update session filepath in job.
+        """
+
+        # 1. Extracts all of the raw data from raw data filepath
+        assert job.raw_filepath, \
+            "Raw data filepath has not been set!"
+        assert job.transform_function, \
+            "Transform function has not been set!"
+        files = os.listdir(job.raw_filepath)
+        files = list(filter(lambda x: x.endswith('.csv'), files))
+        assert len(files) == 1, \
+            "Only supporting one file per dataset folder!"
+        data_filepath = os.path.join(
+                    job.raw_filepath,
+                    files[0]
+                )
+        raw_data = pd.read_csv(data_filepath)
+
+        # 2. Create transformed folder, if it doesn't exist.
+        transformed_filepath = os.path.join(job.raw_filepath, "transformed")
+        if not os.path.isdir(transformed_filepath):
+            os.makedirs(transformed_filepath)
+
+        # 3. Create session folder using timestamp and random characters.
+        new_name = str(uuid.uuid4())
+        session_filepath = os.path.join(transformed_filepath, new_name)
+        os.makedirs(session_filepath)
+
+        # 4. Retrieve transform function and train-test split from job.
+        transform_function = job.transform_function
+        split = job.hyperparams['split']
+
+        # 5. Shuffle raw data, then split into train and test set.
+        transformed_data = transform_function(raw_data)
+        transformed_data = transformed_data.sample(frac=1)
+        split_index = int(len(transformed_data)*split)
+        train = transformed_data.iloc[:split_index] 
+        test = transformed_data.iloc[split_index:]
+
+        # 6. Create train.csv and test.csv in data folder.
+        train.to_csv(
+            os.path.join(session_filepath, 'train.csv'),
+            index=False
+        )
+
+        test.to_csv(
+            os.path.join(session_filepath, 'test.csv'),
+            index=False
+        )
+
+        # 7. Return job with assigned session folder filepath
+        results = DMLResult(
+                    status='successful',
+                    job=job,
+                    results = {
+                        'session_filepath': session_filepath
+                    },
+                    error_message=""
+                )
+        return results 
```

Comment:
How are you ensuring that this session_filepath is actually what will be used throughout the rest of the session? Are you just saying that the session_filepath will remain in the DMLJob, which is persisting inside of the FedAvgOptimizer for the duration of the session?

Reply:
Actually, the `session_filepath` is just returned as a part of `results`. I would assume that in the optimizer somewhere, there's a method like: 

```
def _done_transform_split(self, dmlresult_obj):
...
self.job.session_filepath = dmlresult_obj.results['session_filepath']
...
```

Reply:
Anticipating your followup: (@georgymh can correct me if I'm wrong) but jobs should only be altered outside of the scheduler, since we encountered issues with multithreading and shared memory when we altered the job inside of the Runner.

Reply:
That's fine, I just wanted to make sure that this was your intent because I'll need to modify the optimizer. Matter of fact, if the optimizer isn't doing this currently, how are the tests passing? Since the optimizer isn't calling for a transform_and_split job, and there are assert statements in the train job.

Reply:
See PR description